logic/tool/TypeResolver.java

```diff
@@ -144,355 +83,173 @@ public Conjunction resolveLabels(Conjunction conjunction) {
-                    neighbours.put(resolveRolePlayer, new HashSet<>());
-                    if (hasNoInfo(resolveRolePlayer)) addRootTypeVar(resolveRolePlayer, rootRoleType);
-                    addRelatesConstraint(owner, resolveRolePlayer);
-                    playerType.plays(null, resolveRolePlayer, null);
-                    addNeighbours(playerType, resolveRolePlayer);
-                } else {
-                    playerType.plays(null, roleTypeVar, null);
-                    addNeighbours(playerType, roleTypeVar);
-                }
-            }
-        }
-
-        private void addRelatesConstraint(TypeVariable owner, TypeVariable roleType) {
-            if (owner != null && !owner.reference().isLabel()) owner.relates(roleType, null);
-            if (owner != null) addNeighbours(owner, roleType);
-        }
-
-        private void addRootTypeVar(TypeVariable variable, TypeVariable rootTypeVar) {
-            TypeVariable rootConverted = convert(rootTypeVar);
-            variable.sub(rootConverted, false);
-            addNeighbours(variable, rootConverted);
-        }
-
-        private void convertHas(TypeVariable owner, HasConstraint hasConstraint) {
-            TypeVariable attributeTypeVar = convert(hasConstraint.attribute());
-            owner.owns(attributeTypeVar, null, false);
-            if (hasNoInfo(attributeTypeVar)) addRootTypeVar(attributeTypeVar, rootAttributeType);
-            addNeighbours(owner, attributeTypeVar);
-            assert attributeTypeVar.sub().isPresent();
-            addNeighbours(owner, attributeTypeVar.sub().get().type());
+        private void convertIID(TypeVariable owner, IIDConstraint iidConstraint) {
+            if (conceptMgr.getThing(iidConstraint.iid()) == null)
+                throw GraknException.of(SCHEMATICALLY_UNSATISFIABLE_CONJUNCTION, conjunction);
+            resolverTraversal.labels(owner.id(), conceptMgr.getThing(iidConstraint.iid()).getType().getLabel());
         }
 
         private void convertIsa(TypeVariable owner, IsaConstraint isaConstraint) {
-            TypeVariable isaVar = convert(isaConstraint.type());
-            if (!isaConstraint.isExplicit()) owner.sub(isaConstraint.type(), false);
-            else if (isaConstraint.type().reference().isName()) owner.is(isaConstraint.type());
+            if (!isaConstraint.isExplicit())
+                resolverTraversal.sub(owner.id(), convert(isaConstraint.type()).id(), true);
+            else if (isaConstraint.type().reference().isName())
+                resolverTraversal.equalTypes(owner.id(), convert(isaConstraint.type()).id());
             else if (isaConstraint.type().label().isPresent())
-                owner.label(isaConstraint.type().label().get().properLabel());
-            else throw GraknException.of(ILLEGAL_STATE);
-            addNeighbours(owner, isaVar);
+                resolverTraversal.labels(owner.id(), isaConstraint.type().label().get().properLabel());
```

Comment:
is this if-else if sequence correct? What happens if the user writes a non-explicit labeled isa? will that still work here?

Reply:
can also add the `else ... ILLEGAL_STATE`

Reply:
This is correct. If a user writes a non-explicit labelled `isa`, for example
```
$p isa person;
```

Then it becomes `$p sub person`, which is handled on the traversal engine as:
```
$p --[SUB]-> $_person;
$_person {types:[peson]}
```

Reply:
Agreed it should end with `ILLEGAL_STATE` catch though.